### DISCUSSION TITLE
[Feature Request] Support custom command

### DISCUSSION THREAD

**Discussion by JasperSui** [fastapi/fastapi-cli#82] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched in the documentation/README.
- [X] I already searched in Google "How to do X" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs/README and didn't find an answer.

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```Text
# fastapi <KEYWORD> <COMMAND>
# I'm taking exec as a keyword example
fastapi exec custom-command-1
fastapi exec custom-command-2
```

### Description

I expect this project helps do all CLI stuff like what manage.py in Django does, so here is my feature request for officially supporting custom commands in FastAPI project instead of using own click or typer file to do so.

From my imagination, we will have a place to put our custom commands to run any scripts in shell.

Let me complete the imagination with some example.

Let's say we have a configuration file like `package.json`:

```json
{
  "commands": {
    "custom-command-1": "echo 'I am the 1st custom command'",
    "custom-command-2": "echo 'I am the 2nd custom command'"
  }
}
```

And we can use the following shortcuts in our project:

```bash
fastapi exec custom-command-1 # Equivalent to running `echo 'I am the 1st custom command` in shell.
fastapi exec custom-command-2 # Same as above.
```

This allows us to extend and manage customized commands in one place.

### Operating System

macOS

### Operating System Details

macOS Sonoma 14.1.1

### Python Version

3.12.1

**Comment by mattmess1221** [comment]
Try using [`taskipy`](https://pypi.org/project/taskipy/) to accomplish this.

```toml
[tool.taskipy.tasks]
custom-command-1 = "echo 'I am the 1st custom comman'"
```

```sh
poetry run task custom-command-1
```
